File: Redcoder/Python/4.py
N, M = map(int,input().split())
a = []
for i in range(N):
    aa = [int(i) for i in input().split()]
    a.append(aa)

ans = 0
for i in range(M):
    for j in range(i+1,M):
        count = 0
        for k in range(N):
            count += max(a[k][i], a[k][j])
        ans = max(ans, count)
print(ans)

File: Redcoder/C++/4.cpp
#include <bits/stdc++.h>
#define rep(i,nn,n) for (int i = nn; i < (n); ++i)
using namespace std;
using ll = long long;
using P = pair<int,int>;
#define chmin(x,y) x = min(x,y)
#define chmax(x,y) x = max(x,y)

int N,M;
ll A[100][100];

int main() {
    cin >> N >>M;
    rep(i,0,N)rep(j,0,M){
        cin >> A[i][j];
    }
    ll ans = 0;
    rep(i,0,N)rep(j,i+1,M){
        ll count = 0;
        rep(k,0,N){
            count += max(A[k][j],A[k][i]);
        }
        ans = max(ans,count);
    }

    cout << ans << endl;
    return 0;
}